lib/proxy_util.js

```diff
@@ -0,0 +1,240 @@
+  if (proxyFromEnv.indexOf('://') === -1) {
+    Logger.getInstance().info('Util.getProxyEnv: the protocol was missing from the environment proxy. Use the HTTP protocol.');
+    proxyFromEnv = 'http' + '://' + proxyFromEnv;
+  }
+  proxyFromEnv = new URL(proxyFromEnv);
+  const proxy = {
+    host: Util.validateEmptyString(proxyFromEnv.hostname),
+    port: Number(Util.validateEmptyString(proxyFromEnv.port)),
+    user: Util.validateEmptyString(proxyFromEnv.username),
+    password: Util.validateEmptyString(proxyFromEnv.password),
+    protocol: Util.validateEmptyString(proxyFromEnv.protocol),
+    noProxy: this.getNoProxyEnv(),
+  };
+  this.validateProxy(proxy);
+  return proxy;
+};
+  
+exports.getNoProxyEnv = function () {
+  const noProxy = Util.getEnvVar('no_proxy');
+  if (noProxy) {
+    return noProxy.split(',').join('|');
+  }
+  return undefined;
+};
+
+exports.getHostFromURL = function (destination) {
+  if (destination.indexOf('://') === -1) {
+    destination = 'https' + '://' + destination;
+  }
+  
+  try {
+    return new URL(destination).hostname;
+  } catch (err) {
+    Logger.getInstance().error(`Failed to parse the destination to URL with the error: ${err}. Return destination as the host: ${destination}`);
+    return destination;
+  }
+};
+
+exports.getProxy = function (proxy, fileLocation, isHttps) {
+  if (!proxy && GlobalConfig.isEnvProxyActive()) {
+    proxy = this.getProxyFromEnv(isHttps);
+    if (proxy) {
+      Logger.getInstance().debug(`${fileLocation} loads the proxy info from the environment variable host: ${proxy.host}`);
+    }
+  }
+  return proxy;
+};
+
+exports.getAzureProxy = function (proxy) {
+  const AzureProxy = {
+    ...proxy, host: `${proxy.protocol}${(proxy.protocol).endsWith(':') ? '' : ':'}//${proxy.host}`, 
+  };
+  delete AzureProxy.noProxy;
+  delete AzureProxy.protocol;
+
+  if (!Util.exists(AzureProxy.user) || !Util.exists(AzureProxy.password)) {
+    delete AzureProxy.user;
+    delete proxy.password;
+  }
+  return AzureProxy;
+};
+
+/**
+ * Currently, there is no way to disable loading the proxy information from the environment path in Azure/blob.
+ * To control this proxy option on the driver side, A temporary workaround is hide(remove) the environment proxy from the process
+ * when the client is created (At this time, the client loads the proxy from the environment variables internally). 
+ * After the client is created, restore them with the 'restoreEnvironmentProxy' function.
+ */
+let envProxyList;
+const proxyEnvList = ['http_proxy', 'https_proxy', 'no_proxy'];
+exports.hideEnvironmentProxy = function () {
+  envProxyList = [];
+ 
+  for (const envVar of proxyEnvList) {
+    saveProxyInfoInList(envVar);
+    if (!Util.isWindows()) {
+      saveProxyInfoInList(envVar.toUpperCase());
+    }
+  }
+};
+
+function saveProxyInfoInList(envVar) {
+  envProxyList.push(process.env[envVar]);
+  delete process.env[envVar];
```

Comment:
to avoid future super-hard-to-debug issues, could we please log (at debug loglevel)
* the act of saving the envvar into a list
* especially: the act of deleting the envvar from the runtime environment

Reply:
Done.